bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/device/CommandHandler.java

```diff
@@ -0,0 +1,912 @@
+        X10OnOffCommandHandler(DeviceFeature f) {
+            super(f);
+        }
+
+        @Override
+        public void handleCommand(InsteonChannelConfiguration conf, Command cmd, InsteonDevice dev) {
+            try {
+                byte houseCode = dev.getX10HouseCode();
+                byte houseUnitCode = (byte) (houseCode << 4 | dev.getX10UnitCode());
+                if (cmd == OnOffType.ON || cmd == OnOffType.OFF) {
+                    byte houseCommandCode = (byte) (houseCode << 4
+                            | (cmd == OnOffType.ON ? X10.Command.ON.code() : X10.Command.OFF.code()));
+                    Msg munit = dev.makeX10Message(houseUnitCode, (byte) 0x00); // send unit code
+                    dev.enqueueMessage(munit, feature);
+                    Msg mcmd = dev.makeX10Message(houseCommandCode, (byte) 0x80); // send command code
+                    dev.enqueueMessage(mcmd, feature);
+                    String onOff = cmd == OnOffType.ON ? "ON" : "OFF";
+                    logger.debug("{}: sent msg to switch {} {}", nm(), dev.getAddress(), onOff);
+                }
+            } catch (IOException e) {
+                logger.warn("{}: command send i/o error: ", nm(), e);
+            } catch (FieldException e) {
+                logger.warn("{}: command send message creation error ", nm(), e);
+            }
+        }
+    }
+
+    @NonNullByDefault
+    public static class X10PercentCommandHandler extends CommandHandler {
+        X10PercentCommandHandler(DeviceFeature f) {
+            super(f);
+        }
+
+        @Override
+        public void handleCommand(InsteonChannelConfiguration conf, Command cmd, InsteonDevice dev) {
+            try {
+                //
+                // I did not have hardware that would respond to the PRESET_DIM codes.
+                // This code path needs testing.
+                //
+                byte houseCode = dev.getX10HouseCode();
+                byte houseUnitCode = (byte) (houseCode << 4 | dev.getX10UnitCode());
+                Msg munit = dev.makeX10Message(houseUnitCode, (byte) 0x00); // send unit code
+                dev.enqueueMessage(munit, feature);
+                PercentType pc = (PercentType) cmd;
+                logger.debug("{}: changing level of {} to {}", nm(), dev.getAddress(), pc.intValue());
+                int level = (pc.intValue() * 32) / 100;
+                byte cmdCode = (level >= 16) ? X10.Command.PRESET_DIM_2.code() : X10.Command.PRESET_DIM_1.code();
+                level = level % 16;
+                if (level <= 0) {
+                    level = 0;
+                }
+                houseCode = (byte) x10CodeForLevel[level];
+                cmdCode |= (houseCode << 4);
+                Msg mcmd = dev.makeX10Message(cmdCode, (byte) 0x80); // send command code
+                dev.enqueueMessage(mcmd, feature);
+            } catch (IOException e) {
+                logger.warn("{}: command send i/o error: ", nm(), e);
```

Comment:
This is actually here because Msg.makeMessage() throws an IOException if it can't find the message. This should never happen, except during development. Can I leave as is?

Reply:
done, created custom exception to replace IOException and left as warn.